scripts/avocado

```diff
@@ -31,7 +31,7 @@ def handle_exception(*exc_info):
     # Print traceback if AVOCADO_LOG_DEBUG environment variable is set
     msg = "Avocado crashed:\n" + "".join(traceback.format_exception(*exc_info))
     if os.environ.get("AVOCADO_LOG_DEBUG"):
-        os.write(2, msg + '\n')
+        os.write(2, (msg + '\n').encode('utf-8'))
```

Comment:
Ok, the new syntax is compatible with python3 and python2, but there will be an issue is with python2 when we get non-ascii characters in the message. Solution is to define message as `msg = u"Avocado crashed..."`, I guess.

Reply:
Well the problem I see here is that we can't assume the encoding is `utf-8`. We've had users with non-standard encoding reporting bugs already and as @apahim mentioned we are still negotiating how to properly port Avocado to Py3k as Py2 and Py3 encoding handling is very different.

For example in aexpect we used `os.environ.get("PYTHONENCODING", "utf-8")` which usually works well for Py2 as it's reporting "ascii" even when in "utf-8" mode, but for Py3 it'd be more appropriate to use `sys.getdefaultencoding()`.

I think we should create a function:

```python
if "PYTHONENCODING" in os.environ:
    ENCODING = os.environ["PYTHONENCODING"]
else:
    ENCODING = sys.getdefaultencoding()
    if ENCODING is "ascii" and sys.version <= 2:
        ENCODING = "utf-8"
```

in `avocado.core.output` and use it everywhere. But let's ask @clebergnu who made the most Py3k micro tasks for opinion.